core/corehttp/gateway_handler_unixfs.go

```diff
@@ -3,15 +3,125 @@ package corehttp
 import (
 	"fmt"
 	"html"
+	"io"
 	"net/http"
+	gopath "path"
+	"strconv"
+	"strings"
 	"time"
 
 	files "github.com/ipfs/go-ipfs-files"
+	"github.com/ipfs/go-path/resolver"
+	coreiface "github.com/ipfs/interface-go-ipfs-core"
 	ipath "github.com/ipfs/interface-go-ipfs-core/path"
 	"go.uber.org/zap"
 )
 
-func (i *gatewayHandler) serveUnixFs(w http.ResponseWriter, r *http.Request, resolvedPath ipath.Resolved, contentPath ipath.Path, begin time.Time, logger *zap.SugaredLogger) {
+func (i *gatewayHandler) getOrHeadHandlerUnixfs(w http.ResponseWriter, r *http.Request, begin time.Time, logger *zap.SugaredLogger) {
+	urlPath := r.URL.Path
+	// Only look for _redirects file if we have Origin isolation
+	if hasOriginIsolation(r) {
+		// Check for _redirects file and redirect as needed
+		redirectsFile, err := i.getRedirectsFile(r)
+		if err != nil {
+			switch err.(type) {
+			case resolver.ErrNoLink:
+				// _redirects files doesn't exist, so don't error
+			default:
+				// TODO(JJ): During tests we get multibase.ErrUnsupportedEncoding
+				// This comes from multibase and I assume is due to a fake or otherwise bad CID being in the test.
+				// So for now any errors getting the redirect file are silently ignored.
+				// internalWebError(w, err)
```

Comment:
General question about how we should handle errors reading and parsing the redirects if present.  Also note my comment about issues with test.  I need to dive into this more.